src/nvim/os/unix_defs.h

```diff
@@ -1,7 +1,101 @@
 #ifndef NVIM_OS_UNIX_DEFS_H
 #define NVIM_OS_UNIX_DEFS_H
 
+#include <unistd.h>
+#include <signal.h>
+
+// Defines BSD, if it's a BSD system.
+#ifdef HAVE_SYS_PARAM_H
+# include <sys/param.h>
+#endif
+
+#if defined(DIRSIZ) && !defined(MAXNAMLEN)
+# define MAXNAMLEN DIRSIZ
+#endif
+
+#if defined(NAME_MAX) && !defined(MAXNAMLEN)
+# define MAXNAMLEN NAME_MAX
+#endif
+
+// Default value.
+#ifndef MAXNAMLEN
+# define MAXNAMLEN 512
+#endif
+
+#define BASENAMELEN (MAXNAMLEN - 5)
+
+// Use the system path length if it makes sense.
+#if defined(PATH_MAX) && (PATH_MAX > 1000)
+# define MAXPATHL PATH_MAX
+#else
+# define MAXPATHL 1024
+#endif
+
 #define TEMP_DIR_NAMES {"$TMPDIR", "/tmp", ".", "~"}
 #define TEMP_FILE_PATH_MAXLEN 256
 
+#define HAVE_ACL (HAVE_POSIX_ACL || HAVE_SOLARIS_ACL)
+
+// Special wildcards that need to be handled by the shell.
+#define SPECIAL_WILDCHAR "`'{"
```

Comment:
By defining this you changed the behavior of `gen_expand_wildcards()` on Windows.